platforms/android/library/src/main/java/io/element/android/wysiwyg/utils/HtmlToSpansParser.kt

```diff
@@ -145,6 +158,68 @@ internal class HtmlToSpansParser(
                 text.setSpan(span, start, text.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
                 text.removeSpan(last)
             }
+            "pre" -> {
+                val last = getLast<CodeBlock>() ?: return
+                var start = text.getSpanStart(last)
+                text.removeSpan(last)
+
+                if (start > 0) {
+                    if (text[start] == ZWSP) {
+                        text.replace(start, start + 1, "\n")
+                    } else {
+                        text.insert(start, "\n")
+                    }
+                    start++
+                }
+
+                val codeSpan = CodeBlockSpan(
+                    0xC0A0A0A0.toInt(),
+                    (10 * resourcesProvider.getDisplayMetrics().density).toInt()
+                )
+
+                addZWSP(text.length, isExtra = true)
+
+                text.setSpan(codeSpan, start, text.length, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
+            }
+            "blockquote" -> {
+                val last = getLast<Quote>() ?: return
+                var start = text.getSpanStart(last)
+                text.removeSpan(last)
+
+                if (start > 0) {
+                    if (text[start] == '\u200B') {
+                        text.replace(start, start + 1, "\n")
+                    } else {
+                        text.insert(start, "\n")
+                    }
+                    start++
+                }
+
+                val quoteSpan = QuoteSpan(
+                    indicatorColor = 0xC0A0A0A0.toInt(),
+                    indicatorWidth = (4 * resourcesProvider.getDisplayMetrics().density).toInt(),
+                    indicatorPadding = (6 * resourcesProvider.getDisplayMetrics().density).toInt(),
+                    margin = (10 * resourcesProvider.getDisplayMetrics().density).toInt(),
+                )
+
+                addZWSP(text.length, isExtra = true)
+
+                text.setSpan(quoteSpan, start, text.length, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
+            }
+        }
+    }
+
+    private fun addLineBreak(pos: Int, isExtra: Boolean) {
+        text.insert(pos, "\n")
+        if (isExtra) {
+            text.setSpan(ExtraCharacterSpan(), pos, pos+1, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
+        }
+    }
+
+    private fun addZWSP(pos: Int, isExtra: Boolean) {
+        text.insert(pos, "$ZWSP")
```

Comment:
Actually, it probably should. It was a line break in the beginning, but that led to weird UI issues. 

However, I'm a bit reluctant to move it to the Rust model because:
1. We'll have to re-write most of the tests for quotes/code blocks, and adding this ZWSP after a code block might lead to unexpected situations (issues with backspacing, issues preventing us from merging adjacent nodes, etc.).
2. We'll be removing ZWSPs sooner than later.

I'll give it a try and add the trailing ZWSP next to block nodes, if I see it doesn't break much, I'll move it there.